src/lib.rs

```diff
@@ -404,6 +404,35 @@ pub trait TryIteratorExt: TryIterator {
         Ok(accum)
     }
 
+    /// Consumes the iterator, counting the number of iterations and returns either the
+    /// number of elements or the encountered error.
+    ///
+    /// ```
+    /// use tryiterator::TryIteratorExt;
+    ///
+    /// let values = vec![Ok::<_, i64>(23u64), Ok(40u64)];
+    /// let sum = values.into_iter().try_count();
+    /// assert_eq!(sum, Ok(2));
+    /// ```
+    ///
+    /// ```
+    /// use tryiterator::TryIteratorExt;
+    ///
+    /// let values = vec![Ok::<_, bool>(2u64), Err(false)];
+    /// let sum = values.into_iter().try_count();
+    /// assert_eq!(sum, Err(false));
+    /// ```
+    fn try_count(mut self) -> Result<usize, Self::Error>
+    where
+        Self: Sized,
+    {
+        let mut accum = 0;
+        while let Some(_) = TryIterator::try_next(&mut self)? {
```

Comment:
clippy prefers you use `Option::is_some` rather than the let binding. Other than that, lgtm.

Reply:
not sure I think it's actually more readable, but all praise the clippy!